.dotnet/src/Custom/Batch/CreateBatchOperation.Protocol.cs

```diff
@@ -0,0 +1,273 @@
+    {
+        _pollingInterval ??= new();
+
+        while (!IsCompleted)
+        {
+            PipelineResponse response = GetRawResponse();
+
+            await _pollingInterval.WaitAsync(response, cancellationToken);
+
+            ClientResult result = await GetBatchAsync(_batchId, cancellationToken.ToRequestOptions()).ConfigureAwait(false);
+
+            ApplyUpdate(result);
+        }
+    }
+
+    /// <inheritdoc/>
+    public override void WaitForCompletion(CancellationToken cancellationToken = default)
+    {
+        _pollingInterval ??= new();
+
+        while (!IsCompleted)
+        {
+            PipelineResponse response = GetRawResponse();
+
+            _pollingInterval.Wait(response, cancellationToken);
+
+            ClientResult result = GetBatch(_batchId, cancellationToken.ToRequestOptions());
+
+            ApplyUpdate(result);
+        }
+    }
+
+    internal async Task<CreateBatchOperation> WaitForAsync(ReturnWhen returnWhen, RequestOptions? options)
+    {
+        if (returnWhen == ReturnWhen.Started) return this;
+        await WaitForCompletionAsync(options?.CancellationToken ?? default).ConfigureAwait(false);
+        return this;
+    }
+
+    internal CreateBatchOperation WaitFor(ReturnWhen returnWhen, RequestOptions? options)
+    {
+        if (returnWhen == ReturnWhen.Started) return this;
+        WaitForCompletion(options?.CancellationToken ?? default);
+        return this;
+    }
+
+    private void ApplyUpdate(ClientResult result)
+    {
+        PipelineResponse response = result.GetRawResponse();
+
+        using JsonDocument doc = JsonDocument.Parse(response.Content);
+        string? status = doc.RootElement.GetProperty("status"u8).GetString();
+
+        IsCompleted = GetIsCompleted(status);
+        SetRawResponse(response);
+    }
+
+    private static bool GetIsCompleted(string? status)
+    {
+        return status == InternalBatchStatus.Completed ||
+            status == InternalBatchStatus.Cancelled ||
+            status == InternalBatchStatus.Expired ||
+            status == InternalBatchStatus.Failed;
+    }
+
+    // Generated protocol methods
+
+    /// <summary>
+    /// [Protocol Method] Retrieves a batch.
+    /// </summary>
+    /// <param name="batchId"> The ID of the batch to retrieve. </param>
+    /// <param name="options"> The request options, which can override default behaviors of the client pipeline on a per-call basis. </param>
+    /// <exception cref="ArgumentNullException"> <paramref name="batchId"/> is null. </exception>
+    /// <exception cref="ArgumentException"> <paramref name="batchId"/> is an empty string, and was expected to be non-empty. </exception>
+    /// <exception cref="ClientResultException"> Service returned a non-success status code. </exception>
+    /// <returns> The response returned from the service. </returns>
+    public virtual async Task<ClientResult> GetBatchAsync(string batchId, RequestOptions? options)
```

Comment:
Per arch board discussion: we should remove ids that uniquely identify the operation from protocol method signatures and expose them as string properties on the LRO subclient types.  @KrzysztofCwalina, @tg-msft, @joseharriaga  FYI